### PR TITLE
Feature/ipehr 45

### PR DESCRIPTION
**Pull request by Setest**
API composition create

### CODE REVIEW

File: src/pkg/api/composition.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	}
+
+	defer func(Body io.ReadCloser) {
+		err := Body.Close()
+		if err != nil {
+			log.Panic("Cant close body request")
+		}
+	}(c.Request.Body)
+
+	var request model.CompositionCreateRequest
```
Comment (1db1) [inline src/pkg/api/composition.go:50]: Here we can use model.Composition. CompositionCreateRequest is not required

File: src/pkg/api/composition.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	userId := c.GetString("userId")
+	if userId == "" {
+		c.JSON(http.StatusBadRequest, gin.H{"error": "userId is empty"})
+		return
+	}
+
+	// Checking EHR does not exist
+	_, err = h.service.Doc.EhrsIndex.Get(userId)
+	if errors.Is(err, errors.IsNotExist) {
+		c.AbortWithStatus(http.StatusConflict)
```
Comment (1db1) [inline src/pkg/api/composition.go:71]: This route should not return a 409 error. It should return 404

File: src/pkg/api/composition.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+	var request model.CompositionCreateRequest
+
+	if err = json.Unmarshal(data, &request); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": "Request validation error"})
+		return
+	}
+
+	if !request.Validate() {
+		c.JSON(http.StatusBadRequest, gin.H{"error": "Request validation error"})
```
Comment (1db1) [inline src/pkg/api/composition.go:58]: On validation errors we should return 422

File: src/pkg/docs/service/composition/composition.go [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+func (s CompositionService) MarshalJson(doc *model.Composition) ([]byte, error) {
+	return json.Marshal(doc)
+}
+
+func (s CompositionService) Create(userId string, request *model.CompositionCreateRequest) (*model.Composition, error) {
+	return s.CreateWithId(userId, uuid.New().String(), request) // TODO no new???
+}
+
+func (s CompositionService) CreateWithId(userId, ehrId string, request *model.CompositionCreateRequest) (composition *model.Composition, err error) {
```
Comment (1db1) [inline src/pkg/docs/service/composition/composition.go:45]: The CreateWithId method is not needed, there is no such thing in the specification

File: src/pkg/docs/service/composition/composition.go [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+	}
+	return &doc, nil
+}
+
+func (s CompositionService) MarshalJson(doc *model.Composition) ([]byte, error) {
+	return json.Marshal(doc)
+}
+
+func (s CompositionService) Create(userId string, request *model.CompositionCreateRequest) (*model.Composition, error) {
+	return s.CreateWithId(userId, uuid.New().String(), request) // TODO no new???
```
Comment (1db1) [inline src/pkg/docs/service/composition/composition.go:42]: The new Composition will come directly in the body of the request

File: src/pkg/docs/service/composition/composition.go [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+	// Storage saving
+	docStorageId, err := s.Doc.Storage.Add(docEncrypted)
+	if err != nil {
+		log.Println(err)
+		return err
+	}
+	//
+	// Index EHR userId -> docStorageId
+	if err = s.Doc.EhrsIndex.Add(userId, docStorageId); err != nil {
```
Comment (1db1) [inline src/pkg/docs/service/composition/composition.go:102]: It should not be here, this index is only for ehr
https://track.bsn.si/articles/IPEHR-A-8/EHR-documents-storage-and-indexing-design

File: src/pkg/api/composition.go [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+	service *composition.CompositionService
+}
+
+func NewCompositionHandler(docService *service.DefaultDocumentService, cfg *config.Config) *CompositionHandler {
+	return &CompositionHandler{
+		service: composition.NewCompositionService(docService, cfg),
+	}
+}
+
+func (h CompositionHandler) Create(c *gin.Context) {
```
Comment (1db1) [inline src/pkg/api/composition.go:30]: Need to make a description for swagger similar to other routes

File: src/pkg/api/composition.go (hunk 1 above)
Comment (Setest) [reply src/pkg/api/composition.go]: +

File: src/pkg/api/composition.go (hunk 2 above)
Comment (Setest) [reply src/pkg/api/composition.go:89]: \+

File: src/pkg/api/composition.go (hunk 3 above)
Comment (Setest) [reply src/pkg/api/composition.go:76]: \+

File: src/pkg/docs/service/composition/composition.go (hunk 4 above)
Comment (Setest) [reply src/pkg/docs/service/composition/composition.go]: \+

File: src/pkg/docs/service/composition/composition.go (hunk 5 above)
Comment (Setest) [reply src/pkg/docs/service/composition/composition.go:42]: \+

File: src/pkg/docs/service/composition/composition.go (hunk 6 above)
Comment (Setest) [reply src/pkg/docs/service/composition/composition.go:102]: \\+

File: src/pkg/api/composition.go (hunk 7 above)
Comment (Setest) [reply src/pkg/api/composition.go:48]: \\+

File: src/pkg/api/composition.go (hunk 3 above)
Comment (1db1) [reply src/pkg/api/composition.go:76]: not fixed

**Pull request merged**